### PR TITLE
Reform: Increase Taxable Earnings Subject to Social Security Payroll Taxes

### PR DESCRIPTION
**Pull request by shuyu7zhan**
#3188 

### Summary
:memo::gear::wrench:

This pull request adds a new reform option that increases the taxable earnings for social security payroll taxes, based on the CBO option 6. It adds a new parameter file, a new variable, and a new function to create and apply the reform. It also updates the `reforms.py` file and the changelog.

> _To model a CBO option_
> _We added a new parameter option_
> _We imported a function_
> _That made a construction_
> _Of a reform for social security taxation_

### Walkthrough
*  Add a reform for increasing the taxable earnings for social security payroll taxes ([link](https://github.com/PolicyEngine/policyengine-us/pull/3189/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4), [link](https://github.com/PolicyEngine/policyengine-us/pull/3189/files?diff=unified&w=0#diff-9093e863d515afed5c95f9d0813b76e11830cdda241c362900beafed2a2ecd3bR1-R9), [link](https://github.com/PolicyEngine/policyengine-us/pull/3189/files?diff=unified&w=0#diff-747d1153afbb800168041eec63e9c90cdda566a299986272a07c02b593e4479fR1-R3), [link](https://github.com/PolicyEngine/policyengine-us/pull/3189/files?diff=unified&w=0#diff-cf3902b740190f956e67e07c87560fc66bf9c866bf2224db6a6a70e240ad555dR1-R46), [link](https://github.com/PolicyEngine/policyengine-us/pull/3189/files?diff=unified&w=0#diff-923a45e47f29804b67bd8951728325386f17992b7d1c4dfcea6e784554a47226R7-R9), [link](https://github.com/PolicyEngine/policyengine-us/pull/3189/files?diff=unified&w=0#diff-923a45e47f29804b67bd8951728325386f17992b7d1c4dfcea6e784554a47226R23-R27), [link](https://github.com/PolicyEngine/policyengine-us/pull/3189/files?diff=unified&w=0#diff-923a45e47f29804b67bd8951728325386f17992b7d1c4dfcea6e784554a47226R34))
  - Create a new parameter file `taxable_earnings_upper_threshold.yaml` for the earnings upper threshold subject to social security payroll taxes ([link](https://github.com/PolicyEngine/policyengine-us/pull/3189/files?diff=unified&w=0#diff-9093e863d515afed5c95f9d0813b76e11830cdda241c362900beafed2a2ecd3bR1-R9))
  - Define a new variable `taxable_earnings_for_social_security` that calculates the amount of earnings that are subject to the OASDI FICA tax in `increase_taxable_earnings_for_social_security.py` ([link](https://github.com/PolicyEngine/policyengine-us/pull/3189/files?diff=unified&w=0#diff-cf3902b740190f956e67e07c87560fc66bf9c866bf2224db6a6a70e240ad555dR1-R46))
  - Define a reform class that updates the variable in the model and a helper function that creates the reform based on the parameter values or bypasses the parameter check ([link](https://github.com/PolicyEngine/policyengine-us/pull/3189/files?diff=unified&w=0#diff-cf3902b740190f956e67e07c87560fc66bf9c866bf2224db6a6a70e240ad555dR1-R46))
  - Import the helper function into `reforms.py` and call it within the main function that creates structural reforms from parameters ([link](https://github.com/PolicyEngine/policyengine-us/pull/3189/files?diff=unified&w=0#diff-923a45e47f29804b67bd8951728325386f17992b7d1c4dfcea6e784554a47226R7-R9), [link](https://github.com/PolicyEngine/policyengine-us/pull/3189/files?diff=unified&w=0#diff-923a45e47f29804b67bd8951728325386f17992b7d1c4dfcea6e784554a47226R23-R27))
  - Add the reform to the list of reforms that are composed into a single reform ([link](https://github.com/PolicyEngine/policyengine-us/pull/3189/files?diff=unified&w=0#diff-923a45e47f29804b67bd8951728325386f17992b7d1c4dfcea6e784554a47226R34))
  - Import the function that creates the reform into the `__init__.py` file of the `cbo.payroll` submodule ([link](https://github.com/PolicyEngine/policyengine-us/pull/3189/files?diff=unified&w=0#diff-747d1153afbb800168041eec63e9c90cdda566a299986272a07c02b593e4479fR1-R3))
  - Add a changelog entry for the new reform in `changelog_entry.yaml` ([link](https://github.com/PolicyEngine/policyengine-us/pull/3189/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))

### CODE REVIEW

File: policyengine_us/parameters/gov/contrib/cbo/payroll/taxable_earnings_upper_threshold.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+description: Subject earnings greater than this amount to payroll taxes, in addition to earnings below the maximum taxable amount under current law.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/contrib/cbo/payroll/taxable_earnings_upper_threshold.yaml:1]:
```suggestion
description: The US levies payroll taxes on earnings greater than this amount, in addition to earnings below the maximum taxable amount under current law.
```

File: policyengine_us/parameters/gov/contrib/cbo/payroll/taxable_earnings_upper_threshold.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+values:
+  2021-01-01: .inf
+metadata:
+  unit: USD
+  label: Earnings Upper Threshold Subject to Social Security Payroll Taxes
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/contrib/cbo/payroll/taxable_earnings_upper_threshold.yaml:6]: Let's also rename it `secondary_earnings_threshold.yaml`
```suggestion
  label: Social Security payroll tax secondary earnings threshold
```

File: policyengine_us/reforms/cbo/payroll/increase_taxable_earnings_for_social_security.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        label = "Taxable gross earnings for OASDI FICA"
+        definition_period = YEAR
+        unit = USD
+
+        def formula(person, period, parameters):
+            cap = parameters(period).gov.irs.payroll.social_security.cap
+            upper_threshold = parameters(
+                period
+            ).gov.contrib.cbo.payroll.taxable_earnings_upper_threshold
+            return min_(cap, person("payroll_tax_gross_wages", period)) + max_(
```
Comment (MaxGhenis) [inline policyengine_us/reforms/cbo/payroll/increase_taxable_earnings_for_social_security.py:17]:
```suggestion
            earnings = person("payroll_tax_gross_wages", period)
            p = parameters(period).gov
            base_taxable = min_(earnings, p.irs.payroll.social_security.cap)
            secondary_taxable = max_(0, earnings - p.contrib.cbo.payroll.secondary_earnings_threshold)
            return base_taxable + secondary_taxable
```

File: policyengine_us/reforms/cbo/payroll/increase_taxable_earnings_for_social_security.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+def create_increase_taxable_earnings_for_social_security_reform(
+    parameters, period, bypass: bool = False
+):
+    if bypass:
+        return create_increase_taxable_earnings_for_social_security()
+
+    p = parameters(period).gov.contrib.cbo.payroll
+
+    if p.taxable_earnings_upper_threshold < 9999999999999999999999999:
```
Comment (MaxGhenis) [inline policyengine_us/reforms/cbo/payroll/increase_taxable_earnings_for_social_security.py:36]:
```suggestion
    if p.taxable_earnings_upper_threshold < np.inf:
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/reforms/cbo/payroll/increase_taxable_earnings_for_social_security.py (hunk 4 above)
Comment (shuyu7zhan) [reply policyengine_us/reforms/cbo/payroll/increase_taxable_earnings_for_social_security.py:36]: > 

Hi Max, do we also need to import numpy as np here? At the beginning of this py file?

File: policyengine_us/reforms/cbo/payroll/increase_taxable_earnings_for_social_security.py [hunk 5]
```diff
@@ -1,4 +1,5 @@
 from policyengine_us.model_api import *
+import numpy as np
```
Comment (MaxGhenis) [inline policyengine_us/reforms/cbo/payroll/increase_taxable_earnings_for_social_security.py:2]: Already imported from model_api
```suggestion
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**